joyfunserver/lib/joyfunserver/application.ex

```diff
@@ -0,0 +1,52 @@
+defmodule Joyfunserver.Application do
+
+  @moduledoc false
+  @name __MODULE__
+  
+  use Application
+
+  def start(_type, _args) do
+    import Supervisor.Spec, warn: false
+    children = [
+      worker(Server,[])
+    ]
+
```

Comment:
I don't understand why you didn't choose to make JoyfunServer.agent and Joyfunserver.Mapset.Agent into proper supervised servers with their own API.